splink/comparison_level_library.py

```diff
@@ -86,6 +86,7 @@ def __init__(
         self,
         col_name,
         regex_extract: str = None,
+        set_to_lowercase: bool = False,
```

Comment:
Do you know how this functions when we aren't comparing strings?

Reply:
Good point - I will check this afternoon

Reply:
In duckdb it runs without error - will check in spark now

Reply:
Yea seems to work fine for dates and numerics for both spark and duckdb

Reply:
I'm happy for you to merge this in that case.